src/lib/statistics/statistics_objects/histogram_domain.cpp

```diff
@@ -0,0 +1,133 @@
+#include "histogram_domain.hpp"
+
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+HistogramDomain<pmr_string>::HistogramDomain()
+    :  // Support most of ASCII with maximum prefix length for number of characters.
+      HistogramDomain<pmr_string>(' ', '~', 9) {}
+
+HistogramDomain<pmr_string>::HistogramDomain(const char min_char, const char max_char, const size_t prefix_length)
+    : min_char(min_char), max_char(max_char), prefix_length(prefix_length) {
+  Assert(min_char <= max_char, "Invalid character range");
+  Assert(prefix_length > 0, "String prefix too short");
+}
+
+size_t HistogramDomain<pmr_string>::character_range_width() const {
+  // One cast at a time to suppress gcc warnings: https://stackoverflow.com/a/27513865
+  return static_cast<size_t>(static_cast<unsigned char>(max_char - min_char + 1));
+}
+
+HistogramDomain<pmr_string>::IntegralType HistogramDomain<pmr_string>::string_to_number(
+    const pmr_string& string_value) const {
+  // The prefix length must not overflow for the number of supported characters when representing strings as numbers.
+  DebugAssert(prefix_length < std::log(std::numeric_limits<uint64_t>::max()) / std::log(character_range_width() + 1),
+              "String prefix too long");
+  if (!contains(string_value)) {
+    return string_to_number(string_to_domain(string_value));
+  }
+
+  auto base = _base_number();
+  auto value = IntegralType{0};
+
+  for (auto idx = size_t{0}; idx < std::min(string_value.size(), prefix_length); ++idx) {
+    value += (string_value[idx] - min_char) * base + 1;
+    base -= ipow(character_range_width(), prefix_length - idx - 1);
+  }
+
+  // If `value` is longer than `prefix_length` add 1 to the result.
+  // This is required for the way EqualWidthHistograms calculate bin edges.
+  value += string_value.length() > prefix_length ? 1 : 0;
+
+  return value;
+}
+
+pmr_string HistogramDomain<pmr_string>::string_to_domain(const pmr_string& string_value) const {
+  auto converted = string_value;
+  for (auto pos = size_t{0}; pos < converted.size(); ++pos) {
+    converted[pos] = std::min(max_char, std::max(min_char, converted[pos]));
+  }
+  return converted;
+}
+
+bool HistogramDomain<pmr_string>::contains(const pmr_string& string_value) const {
+  for (auto pos = size_t{0}; pos < string_value.size(); ++pos) {
+    if (string_value[pos] > max_char || string_value[pos] < min_char) {
+      return false;
+    }
+  }
+  return true;
+}
+
+pmr_string HistogramDomain<pmr_string>::next_value(const pmr_string& string_value) const {
+  DebugAssert(contains(string_value), "Unsupported character, cannot compute next_value()");
```

Comment:
What happens if I search for `s LIKE a☺a%`?

Reply:
As discussed, the emoji is previously clamped to the supported character range. renamed the parameter to "string_in_domain" and dc'ed it.